Floofbot/Modules/TagCommands.cs

```diff
@@ -0,0 +1,206 @@
+﻿using Discord;
+using Discord.Commands;
+using Floofbot.Services.Repository;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Tag = Floofbot.Services.Repository.Models.Tag;
+
+namespace Floofbot.Modules
+{
+    [Group("tag")]
+    [Summary("Tag commands")]
+    [RequireContext(ContextType.Guild)]
+    [RequireUserPermission(Discord.GuildPermission.AttachFiles)]
+    public class TagCommands : ModuleBase<SocketCommandContext>
+    {
+        private static readonly Discord.Color EMBED_COLOR = Color.Magenta;
+        private static readonly int TAGS_PER_PAGE = 50;
+        private static readonly List<string> SUPPORTED_IMAGE_EXTENSIONS = new List<string>
+        {
+            "jpg", "png", "jpeg", "webp", "gifv", "gif", "mp4"
+        };
+
+        private FloofDataContext _floofDb;
+
+        public TagCommands(FloofDataContext floofDb)
+        {
+            _floofDb = floofDb;
+        }
+
+        [Command("add")]
+        [Priority(0)]
```

Comment:
is the priority tag needed if we are overloading the function anyway?

Reply:
Just tested it. We need [Priority(1)] on the overload but we don't need [Priority(0)] here. I'll edit it out.